### DISCUSSION TITLE
List all valid options for a single child

### DISCUSSION THREAD

**Discussion by stefanmeili** [jim-easterbrook/python-gphoto2#169] [Q&A] [answered]
I'm hoping to find a way to get a list of all valid options for a configuration option.

For example, I can list the current setting for iso using:
```python
camera = gp.Camera()
config = camera.get_config()
current_iso = config.get_child_by_name('iso').get_value()
```

How can I list all valid options for my camera:
`[50,72,100,140,200,280,400,560,800,1100,1600,2200,3200,4500,6400,9000,12800,18000,25600,36000,51200]`

Without resorting to parsing `camera.get_summary()`?

**Comment by jim-easterbrook** [comment, marked as answer]
`gp_camera_get_single_config` gets a single config widget, `gp_widget_count_choices` tells you how many options the widget has, `gp_widget_get_choice` gets each of the options.

**Reply by stefanmeili** [reply]
Thanks - that got me there.
```python
def get_param_range(setting):
    camera = gp.Camera()
    try:
        n_choices = camera.get_single_config(setting).count_choices()
        single_config = camera.get_single_config(setting)
        setting_options = [single_config.get_choice(i) for i in range(n_choices)]
    except:
        raise ValueError(f'invalid or no selections available for for setting "{setting}""')
    camera.exit()
    return setting_options
```

**Reply by jim-easterbrook** [reply]
Repeatedly calling `get_single_config` is suboptimal. Better to call it once to get the "widget", then call `count_choices` and `get_choice` on that widget.

**Reply by stefanmeili** [reply]
Good point. Updated.
